enhancements/update/targeted-update-edge-blocking.md

```diff
@@ -0,0 +1,966 @@
+    Any targeting of the blocked edges will reduce the number of blocked clusters which would have not been impacted, and thus reduce the contention between protecting vulnerable clusters and inconveniencing invulnerable clusters, so even overly broad scoping is better than no scoping at all.
+* Specifying a particular update service implementation.
+    This enhancement floats some ideas, but the details of the chosen approach are up to each update service's maintainers.
+
+## Proposal
+
+### Enhanced graph-data schema for blocking edges
+
+[The existing blocked-edges schema][block-edges] will be extended with the following new properties:
+
+* `url` (optional, [string][json-string]), with a URI documenting the blocking reason.
+    For example, this could link to a bug's impact statement or knowledge-base article.
+* `reason` (optional, [string][json-string]), with a CamelCase reason suitable for [a `ClusterOperatorStatusCondition` `reason` property][api-reason].
+* `message` (optional, [string][json-string]), with a human-oriented message describing the blocking reason, suitable for [a `ClusterOperatorStatusCondition` `message` property][api-message].
+* `matchingRules` (optional, [array][json-array]), defining conditions for deciding which clusters have the update recommended and which do not.
+  The array is ordered by decreasing precedence.
+  Consumers should walk the array in order, and stop after the first rule they can successfully evaluate.
+  If no condition can be successfully evaluated, the update should not be recommended.
+  Each entry must be an [object][json-object] with at least the following property:
+  * `type` (required, [string][json-string]), defining the type in [the condition type registry](#condition-type-registry).
+    For example, `type: promql` identifies the condition as [the `promql` type](#promql).
+  Additional properties for each entry are defined in [the cluster-condition type registry](#cluster-condition-type-registry).
+
+[The schema version][graph-data-schema-version] would also be bumped to 1.1.0, because this is a backwards-compatible change.
+Consumers who only understand graph-data schema 1.0.0 would ignore the `matchingRules` property and block the edge for all clusters.
+The alternative of failing open is discussed [here](#failing-open).
+
+### Cluster-condition type registry
+
+This registry contains the set of known cluster-condition types, and will be extended if and when future enhancements propose new types.
+
+#### promql
+
+The `type: promql` condition entry is an [object][json-object] with the following additional property:
+
+* `promql` (optional, [string][json-string]), with a [PromQL][] query classifying clusters.
+  This query will be evaluated on the local cluster, so it has access to data beyond the subset that is [uploaded to Telemetry][uploaded-telemetry].
+  The query should return a 1 in the happy case (update should be allowed) and a 0 in the sad case (update should be blocked).
```

Comment:
If the value is optional, how should it be interpreted if it is empty?

Reply:
Empty is "failed to evaluate".  I've pushed af6d5bb -> 3c5ce31 to call that out explicitly, and link to the existing alternatives section.